Products/Zuul/catalog/model_catalog_init.py

```diff
@@ -320,13 +322,32 @@ def run(processor_count=8, hard=False):
     log.info("Number of objects indexed: {0}".format(total_count))
 
 
-def reindex_model_catalog(root="/zport", idxs=None, processor_count=8, hard=False):
+def reindex_model_catalog(dmd, root="/zport", idxs=None):
     """
-    @param  root: zodb node to reindex
-    @params idxs: fields to index
+    Performs a single threaded soft reindex
     """
-    # @TODO pass root and idxs when implemented
-    run(processor_count, hard)
+    start = time.time()
+    log.info("Performing soft reindex on model_catalog. Params = root:'{}' / idxs:'{}'".format(root, idxs))
+    modelindex = init_model_catalog()
+    uids = get_uids(modelindex, root=root)
+    if uids:
+        index_updates = []
+        for uid in uids:
+            try:
+                obj = dmd.unrestrictedTraverse(uid)
+            except KeyError:
```

Comment:
I'm not what the scenario is where it happens, but I believe `dmd.unrestrictedTraverse` can also throw `zExceptions.NotFound` when a path is not found in ZODB